### DISCUSSION TITLE
What is the recommended way to customize the dict and json output?

### DISCUSSION THREAD

**Discussion by dmsfabiano** [pydantic/pydantic#5300] [Question]
Say I have a simple use-case like

```python
from pydantic import BaseModel

class ParentModel(BaseModel):
  pass

class ChildModel(ParentModel):
  field_one: str
  field_two: int
```

what I would like to do is for my `json` and `dict` or any serialization and deserialization to include the `type` of the field, and I would prefer for that to be implemented in the parent and leveraged by all the children

current output 
```
child = ChildModel("test_field_one", 3)
child.dict()
{"field_one": "test_field_one", "field_two": 3}
```
desired output
```
child = ChildModel("test_field_one", 3)
child.dict()
{"field_one": {"value": "test_field_one", "type": "str"}, "field_two": {"value": 3, "type": "int"}}
```
Now, of course, i want to be able to create a new `ChildModel` from the output of `dict`. What is the most reliable way to build an encoder and decoder? I have read through the documentation [here](https://docs.pydantic.dev/usage/exporting_models/), but I don't see many options:

- Overwrite `dict` and `json` to include the type, by iterating through the fields? overwrite the `__dict__` method? But then, how would a make a custom loader?
- I saw the concept of the `json_encoder` however those are specific to the type and not all the fields
- create a custom `export` and `import` function that parses the custom dict (this is my least preferred)

**Comment by raceychan** [comment]
I would go for option 1 and also a customized parser, use parse_obj when constructing from dict
